node/utils/attributes.ts

```diff
@@ -0,0 +1,181 @@
+    to: string
+    active: boolean
+  }[]
+}
+
+interface TextAttribute {
+  type: 'text'
+  values: {
+    key: string
+    count: number
+    active: boolean
+    label: string
+  }[]
+}
+
+type FilterType = 'PRICERANGE' | 'TEXT'
+interface Filter {
+  type: FilterType
+  name: string
+  hidden: boolean
+  values: FilterValue[]
+}
+
+interface FilterValue {
+  quantity: number
+  name: string
+  key: string
+  value: string
+  selected?: boolean
+  range?: {
+    from: number
+    to: number
+  }
+}
+
+/**
+ * Convert from Biggy's attributes into Specification Filters.
+ *
+ * @export
+ * @param {Attribute[]} [attributes] Attributes from Biggy's API.
+ * @returns {Filter[]}
+ */
+export const attributesToFilters = ({
+  total,
+  attributes,
+}: {
+  total: number
+  attributes?: Attribute[]
+}): Filter[] => {
+  if (either(isNil, isEmpty)(attributes)) {
+    return []
+  }
+
+  return attributes!.map(attribute => {
+    const { type, values } = convertValues(attribute, total)
+
+    return {
+      values,
+      type,
+      name: attribute.label,
+      hidden: !attribute.visible,
+    }
+  })
+}
+
+/**
+ * Convert values, and create FilterType, that can be either `PRICERANGE` or
+ * `TEXT`, only price uses the `PRICERANGE` type, other number attributes that
+ * come from Biggy's API are transformed into TEXT filters.
+ *
+ * Location attributes are also transformed in TEXT filters, but when a location
+ * is selected, only a single bucket is returned (the API returns multiple buckets,
+ * but does not return the selected one, and there's no need to select multiple
+ * location buckets).
+ *
+ * If an attribute is not of type `'text' | 'number' | 'location'`, an error will
+ * be thrown.
+ *
+ * @param {Attribute} attribute
+ * @returns {{ type: FilterType; values: FilterValue[] }}
+ */
+const convertValues = (
+  attribute: Attribute,
+  total: number
+): { type: FilterType; values: FilterValue[] } => {
+  // When creating a filter for price attribute, it should be the only one to use
+  // the type `'PRICERANGE'`.
+  if (attribute.type === 'number' && attribute.key === 'price') {
+    return {
+      type: 'PRICERANGE',
+      values: [
+        {
+          quantity: attribute.values.reduce(
+            (acum: number, value: any) => acum + value.count,
+            0
+          ),
+          name: unescape(attribute.label),
+          key: attribute.key,
+          value: attribute.key,
+          range: {
+            from: attribute.minValue,
+            to: attribute.maxValue,
+          },
+        },
+      ],
+    }
+  }
+
+  // For other `number` and `location` attributes we use TEXT based filters (checkboxes),
+  // with buckets (e.g.: 0 - 30, 30 - 90, etc).
+  if (attribute.type === 'number' || attribute.type === 'location') {
+    const valuePrefix = attribute.type === 'location' ? 'l:' : ''
+
+    // When a bucket is active, we should only show it, and none of the othter options.
+    if (attribute.active) {
```

Comment:
can you test this on `samsungbr` to make sure it will work for the `polegadas` filter? :)

Reply:
Yes! It's working here: [Workspace](https://christian--samsungbr.myvtex.com/tv?_q=tv&map=category-2).